plugins/item_licenses/web_client/js/setup.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Show license on the item page.
+ */
+girder.wrap(girder.views.ItemView, 'render', function (render) {
+    // ItemView is a special case in which rendering is done asynchronously,
+    // so we must listen for a render event.
+    this.once('g:rendered', function () {
+        var itemLicenseItemWidget = new girder.views.item_licenses_ItemWidget({ // eslint-disable-line new-cap
+            item: this.model,
+            parentView: this
+        }).render();
+
+        $('.g-item-info').append(itemLicenseItemWidget.el);
+    }, this);
+
+    render.call(this);
+
+    return this;
+});
+
+/**
+ * Allow selecting license when uploading an item.
+ */
+girder.wrap(girder.views.HierarchyWidget, 'uploadDialog', function (uploadDialog) {
+    girder.restRequest({
```

Comment:
This means that the pop-up and rendering of the upload dialog is completely blocked until this REST request resolves.

Ideally, I'd hope to see the dialog render immediately, and the dropdown with license options get populated asynchronously.